src/client/index.ts

```diff
@@ -18,111 +79,268 @@ export interface NotificationOptions {
+    category: string;
+
     /**
-     * URL of the icon to be displayed in the Notification.
+     * URL of the icon to be displayed in the notification.
      */
     icon?: string;
+
     /**
-     * Any custom context data associated with the Notification.
+     * Any custom context data associated with the notification.
      */
     customData?: CustomData;
+
     /**
-     * The timestamp shown on the Notification.  This is presentational only - a future date will not incur a scheduling action.
+     * The timestamp shown on the notification. If omitted, the current date/time will be used.
+     *
+     * This is presentational only - a future date will not incur a scheduling action.
      */
     date?: Date;
+
     /**
-     * Text and icons for up to two Notification action buttons.
+     * A timestamp at which point the notification will be removed from the user's notification center.
+     *
+     * If omitted or `null`, notification will persist until user explicitly actions/clears it.
      */
-    buttons?: OptionButton[];
-}
+    expires?: Date|null;
 
-/**
- * Configuration options for constructing a Button within a Notification.
- */
-export interface OptionButton {
-    title: string;
-    iconUrl?: string;
+    /**
+     * A list of button definitions, to display below the notification text.
+     *
+     * Notifications support up to four buttons, attempting to add more than four will result in the notification
+     * being rejected by the service.
+     */
+    buttons?: ButtonOptions[];
+
+    /**
+     * An action that will fire when the notification is clicked. This only fires on clicks to the notification body,
+     * interactions with buttons (both application-defined buttons, and the default 'X' close button) will not trigger
+     * a select action.
+     *
+     * If omitted or `null`, applications will not receive a {@link NotificationActionEvent|`notification-action`}
+     * event when the notification is clicked. See {@link Actions} for more details on Notification actions, and
+     * receiving interaction events from notifications.
+     */
+    onSelect?: ActionDeclaration<never, never>|null;
 }
 
 /**
- * User-defined context data that can be attached to Notifications.
+ * Application-defined context data that can be attached to notifications.
  */
 export type CustomData = any;
 
 /**
- * A fully hydrated form of the {@link NotificationOptions}.
+ * A fully-hydrated form of {@link NotificationOptions}.
+ *
+ * After {@link create|creating} a notification, the service will return this object - which will be the given options
+ * object, with any unspecified fields filled-in with default values.
+ *
+ * This object should be treated as immutable, modifying its state will not have any effect on the notification that
+ * the user sees on-screen.
  */
-export type Notification = Required<NotificationOptions>;
+export type Notification = Readonly<NotificationOptions & Required<NotificationOptions>>;
```

Comment:
Was previously some funky "`Omit` and intersect" stuff here. Didn't do a great job of cleaning that up :-)

Reply:
That was the stuff I added around the date transport stuff, since Options takes a `Date` but we pass a string (or number, I forget) over the bus and use that internally. Do we have a different way of dealing with that discrepancy now?

Reply:
Yeah, that's encapsulated in the event deserialisation mechanism - see [EventRouter.ts](https://github.com/HadoukenIO/notifications-service/pull/83/files#diff-2d4b539683938fa35f883aefd0944c20R33-R35)/[internal.ts](https://github.com/HadoukenIO/notifications-service/pull/83/files#diff-a73489a46aa1b827a02dcf40199ac582R49-R54) for the type mappings, and [client/index.ts](https://github.com/HadoukenIO/notifications-service/pull/83/files#diff-31df54a9a51bd3da26df362832cc1c9fR19-R46) for the runtime conversion stuff.

Reply:
Ah of course. Never mind then.